src/Plugin/Journal.php

```diff
@@ -0,0 +1,31 @@
+<?php
+
+namespace Http\Client\Common\Plugin;
+
+use Http\Client\Exception;
+use Psr\Http\Message\RequestInterface;
+use Psr\Http\Message\ResponseInterface;
+
+/**
+ * Records history of HTTP calls.
+ *
+ * @author Joel Wurtz <joel.wurtz@gmail.com>
+ */
+interface Journal
```

Comment:
or to its own repository? because its rather specific and a whole bunch
of files. and will rarely be used outside a framework integration like
symfony. moving it to message feels a bit random too, no? but journal
interfaces here in client feels odd.